### ISSUE TITLE
Current status

### ISSUE THREAD

**Issue by mdpedersen** [Spreads/Spreads#164]
Hello,

Thank you for this library, it looks really good.

However, I am struggling to get started with even very basic applications. The provided examples seem out of date, and documentation is lacking. For example, `Series.Do` and `Series.Scan` don't seem to exist in the current release. Apologies if there something obvious I am missing here, but I would love to be able to give Spreads a proper go.

I'm on Mac OS, using FSharp and the 2019.0.415 Spreads build via NuGet.

Maybe this is related, but if I include the `Spreads.extensions` package, I get errors complaining that references to the type `Spreads.ISeries2` in assembly `Spreads.collections` are found but the type is not. I'm not sure if extensions is actually needed.
 
Any suggestions much appreciated!

**Comment by buybackoff** [comment]
Hi! Thanks a lot for your interest in the library!

Sorry if the releases page or version number make you think it's ready to use. The entire library is in the process of overhaul to make it stable (changing interfaces after lessons learned, significantly improving performance due to algorithmic changes). And I haven't updated the Readme since initial release because it was like shooting at a quickly moving target. Auto generated API docs are here: http://docs.dataspreads.io/spreads/api/index.html

`Series.Do` is now not relevant because all series are async streams that are consumable via `await foreach` in C# 8.0. If you are in F# or older C# then at the moment you have to use `series.GetCursor` or `series.GetAsyncCursor()` manually, I will add back `series.Do/ForEach` as extension methods later. `series.Scan` requires more work because it requires caching all intermediate values, which requires lots of memory.

> Maybe this is related, but if I include the Spreads.extensions package

`Spreads.Extensions` is an old package, not used anymore. For end users only the top-level `Spreads` package is the intended entry point. For sub-packages you should know internals and why you are using it, and not the top level one (e.g. there are lots of useful performance/infrastructure/serialization stuff in `Spreads.Core` that are general purpose).

See #163 on why I released a version without a suffix so that NuGet considers it "stable". I think the infrastructure parts of `Spreads.Core` are in more or less `Release candidate` state now. `SortedMap` is a battle-tested container but I'm deprecating it and rewriting everything in C# with a significant performance boost.

I was focused on the persistence layer over the past several months and I published an alpha version here: https://github.com/DataSpreads/DataSpreads (despite 9 visible commits it's actually 247 and several months of work, couldn't publish the commit history). Now I'm making it stable with minimum features and will return to this library after that. I know what to do here to make this repo stable, but now it's more like a physical labor of actually typing the code & tests.

The bottom line is that if you are not ready to dive into the internals of this library, I have to kindly ask you to come back later. Or subscribe to announcements here: http://docs.dataspreads.io/. That stable-everything-working day will eventually come, but I'm alone doing all that stuff.

**Comment by mdpedersen** [comment]
Thank you very much for the detailed explanation! I'll be following this with interest.

**Comment by buybackoff** [comment]
@mdpedersen BTW, if you could share your use case that will help a lot!

**Comment by mdpedersen** [comment]
Use case is for algorithmic trading.

**Comment by buybackoff** [comment]
Current status: https://twitter.com/buybackoff/status/1252687740006813697
